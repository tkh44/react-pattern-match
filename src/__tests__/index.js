/* eslint-disable jsx-quotes */
/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import Match from '../index'

function Box (props) {
  return <div {...props} />
}

describe('react-pattern-match', () => {
  test('renders', () => {
    const tree = renderer
      .create(
        <Match
          value={4}
          render={eq => [
            eq(4, () => <Box name="int" />),
            eq(() => 4, () => <Box name="fn" />)
          ]}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('value is fn', () => {
    function getValue () {
      return ['a', 'b', 'c']
    }

    const tree = renderer
      .create(
        <Match
          value={getValue}
          render={eq => [
            eq(['a', 'b', 'c'], () => <Box name="should match array" />),
            eq(['a', 'b', 'c', 'd'], () =>
              <Box name="should not match array" />
            ),
            eq(() => ['a', 'b', 'c'], () => <Box name="should match fn" />),
            eq(
              () => ['a', 'b', 'c', 'd'],
              () => <Box name="should not match fn" />
            )
          ]}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
