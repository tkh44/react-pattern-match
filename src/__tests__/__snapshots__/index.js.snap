// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-pattern-match renders 1`] = `
Array [
  <div
    name="int"
/>,
  <div
    name="fn"
/>,
]
`;

exports[`react-pattern-match value is fn 1`] = `
Array [
  <div
    name="should match array"
/>,
  <div
    name="should match fn"
/>,
]
`;
